kernel_gateway/services/kernels/manager.py

```diff
@@ -76,7 +76,16 @@ def start_kernel(self, *args, **kwargs):
                 client = kernel.client()
                 # Only start channels and wait for ready in HTTP mode
                 client.start_channels()
-                client.wait_for_ready()
+                try:
+                    client.wait_for_ready()
```

Comment:
With [wait_for_ready](https://github.com/jupyter/jupyter_client/blob/637fb55afe9989b4a5f420d5e32a5a11e907f770/jupyter_client/blocking/client.py#L23) it is synchronous.  We could then init the client and check that it has started/is ready regardless of whether it has a seed_source (missed that part).  I would have thought the approach in the first commit would work though, as I think its better not to deal with the problem on this level.